genie-common/src/main/java/com/netflix/genie/common/dto/JobRequest.java

```diff
@@ -93,6 +97,17 @@
         this.memory = builder.bMemory;
         this.timeout = builder.bTimeout;
         this.applications = ImmutableList.copyOf(builder.bApplications);
+        this.grouping = builder.bGrouping;
+        this.groupingInstance = builder.bGroupingInstance;
+    }
+
+    /**
+     * Get the arguments to be put on the command line along with the command executable.
+     *
+     * @return The command arguments
+     */
+    public Optional<String> getCommandArgs() {
```

Comment:
We can abstract out this command args logic into a separate base class.

Reply:
ugh. I hate that you're right about this.

Reply:
After looking at it the way the inheritance structure for the DTO's are setup this isn't really possible since `Job` extends from `CommonDTO` and `JobRequest` from `ExecutionEnvironmentDTO`. If I make `ExecutionEnvironmentDTO` extend a `JobCommonDTO` it then screws up `Application`, `Cluster`, `Command` cause they all extend that `ExecutionEnvironmentDTO`. Since Java doesn't support multiple inheritance this probably isn't worth the hassle right now. For 4.0 we'll clean all this up anyway and deprecate or remove all these DTO's in favor of new ones that line up to 4.0 way of doing things whatever that ends up being.